### PR TITLE
Keep `sample` in TQL2 around for backwards compat

### PR DESCRIPTION
**Pull request by dominiklohmann**

### CODE REVIEW

**Review by jachris — APPROVED** [review]

File: libtenzir/builtins/aggregation-functions/sample.cpp [hunk 1]
```diff
@@ -67,6 +68,13 @@ class plugin : public virtual aggregation_function_plugin {
   auto aggregation_default() const -> data override {
     return caf::none;
   }
+
+  auto make_aggregation(invocation inv, session ctx) const
+    -> failure_or<std::unique_ptr<aggregation_instance>> override {
+    const auto* first = plugins::find<aggregation_plugin>("first");
+    TENZIR_ASSERT(first);
+    return first->make_aggregation(invocation{inv.call}, ctx);
```
Comment (jachris) [inline libtenzir/builtins/aggregation-functions/sample.cpp:76]: Actually, would you mind adding a deprecation warning here?

Comment (dominiklohmann) [reply libtenzir/builtins/aggregation-functions/sample.cpp:76]:
```diff
diff --git a/libtenzir/builtins/aggregation-functions/sample.cpp b/libtenzir/builtins/aggregation-functions/sample.cpp
index a8341b5b69f..0f4c23e2968 100644
--- a/libtenzir/builtins/aggregation-functions/sample.cpp
+++ b/libtenzir/builtins/aggregation-functions/sample.cpp
@@ -71,6 +71,9 @@ class plugin : public virtual aggregation_function_plugin,
 
   auto make_aggregation(invocation inv, session ctx) const
     -> failure_or<std::unique_ptr<aggregation_instance>> override {
+    diagnostic::warning("deprecated: use `first` instead")
+      .primary(inv.call)
+      .emit(ctx);
     const auto* first = plugins::find<aggregation_plugin>("first");
     TENZIR_ASSERT(first);
     return first->make_aggregation(invocation{inv.call}, ctx);
```

This revealed an interesting bug: parse-time diagnostics from aggregation functions always get emitted twice. I don't think that should be the case.

Comment (jachris) [reply libtenzir/builtins/aggregation-functions/sample.cpp:76]: Oh, yes. That's a regression from https://github.com/tenzir/tenzir/pull/4527, caused by moving the deduplicator into the inner execution function.

**Comment by dominiklohmann** [comment]
We just went with using `first` in the Tenzir Platform now, so I'm closing this without merging.

**Pull request closed without merging**